cmd/ar+https/main.go

```diff
@@ -0,0 +1,386 @@
+package main
+
+import (
+	"bufio"
+	"context"
+	"crypto/md5"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+)
+
+func main() {
+	apt := NewAptMethod(os.Stdout, bufio.NewReader(os.Stdin))
+	apt.Run()
+}
+
+func (m *AptMethod) Run() {
+	m.writer.SendCapabilities()
+	for {
+		msg, err := m.reader.ReadMessage()
+		if err != nil {
+			m.writer.Log(fmt.Sprintf("Error reading APT message: %v", err))
+			continue
+		}
+		switch msg.Code {
+		case 600:
+			m.handleAcquire(msg)
+		case 601:
+			m.handleConfigure(msg)
+		default:
+			m.writer.Fail("Unsupported API")
+		}
+	}
+}
+
+type AptMethod struct {
```

Comment:
Why is code duplicated in this file (e.g. `AptMethod`, `AptMessage`, readers, writers...)?

Reply:
developer error haha